go/billing/api.py

```diff
@@ -118,44 +126,42 @@ def _create_notification_mapping(self):
 
     def render_POST(self, request):
         """Handle an HTTP POST request"""
-        data = self._parse_json(request)
-        if data:
-            account_number = data.get('account_number', None)
-            message_id = data.get('message_id', None)
-            tag_pool_name = data.get('tag_pool_name', None)
-            tag_name = data.get('tag_name', None)
-            provider = data.get('provider', None)
-            message_direction = data.get('message_direction', None)
-            session_created = data.get('session_created', None)
-            transaction_type = data.get('transaction_type', None)
-
-            if all((account_number, message_id, tag_pool_name, tag_name,
-                    message_direction, session_created is not None)):
-                d = self.create_transaction(
-                    account_number, message_id, tag_pool_name, tag_name,
-                    provider, message_direction,
-                    session_created, transaction_type)
-
-                d.addCallbacks(self._render_to_json, self._handle_error,
-                               callbackArgs=[request], errbackArgs=[request])
-            else:
-                self._handle_bad_request(request)
-        else:
+        data = self._parse_post(request)
+
+        if data is None:
             self._handle_bad_request(request)
+        else:
+            d = self.create_transaction(*pluck(data, self.FIELDS))
+
+            d.addCallbacks(self._render_to_json, self._handle_error,
+                           callbackArgs=[request], errbackArgs=[request])
+
         return NOT_DONE_YET
 
+    def _parse_post(self, request):
+        data = self._parse_json(request) or {}
+        data = dict((k, data.get(k)) for k in self.FIELDS)
+        fields = set(self.FIELDS) - set(self.NULLABLE_FIELDS)
```

Comment:
Perhaps we should put ``NON_NULLABLE_FIELDS = tuple(sorted(set(FIELDS) - set(NULLABLE_FIELDS)))`` on the class instead of calculating this each time?